packages/nocodb/src/lib/meta/api/attachmentApis.ts

```diff
@@ -16,26 +16,33 @@ import Local from '../../v1-legacy/plugins/adapters/storage/Local';
 import { NC_ATTACHMENT_FIELD_SIZE } from '../../constants';
 
 const isUploadAllowed = async (req: Request, _res: Response, next: any) => {
-  if (!req['user']?.id) {
+  if (!req['user']) {
```

Comment:
The test case is still failing. I believe you can simply do in this way.

```ts
  if (
    !req['user']?.id ||
    (req['user']?.isPublicBase &&
      !req['user'].roles?.includes(ProjectRoles.EDITOR))
  ) {
    NcError.unauthorized('Unauthorized');
  }

```

Reply:
Not quite.. Then it fails if *either* the id is missing or the public url & role. But the success case is that *both* the id is missing as well as the public url & role is are there.

Reply:
Lemme push something to test.